### ISSUE TITLE
"must only be invoked from the spec file or support file." errors using cy.selectFile() & cy.readFile()

### ISSUE THREAD

**Issue by S-Lomax** [cypress-io/cypress#27457]
### Current behavior

Recently updated to Cypress 12.17.3 from 12.14 and are still getting "must only be invoked from the spec file or support file." errors when using commands `cy.selectFile()`, `cy.readFile()` and `cy.task()`.

Exactly the same behaviour as [https://github.com/cypress-io/cypress/issues/27149](url)

### Test code to reproduce

```
cy.get('[data-cy="QA_Automation_Section_Zip-accordion-button"]').click()
cy.get('[data-cy="thumbnail-button-QA_Automation_Slide_Zip"]').click()
cy.get('[data-cy="upload-container"]').selectFile('cypress/fixtures/QA_Automation_Slide_Deploy.png', {force: true})
```

### Cypress Version

12.17.3

### Node version

18.16.1

### Operating System

Windows 10.0.19045.3208

**Comment by nagash77** [comment]
Hi @S-Lomax, In 12.15 we did do some work around privileged commands.  However your example does not give enough context into how you are calling these functions that would cause this problem.  In order to give our engineers the best chance at recreating your problem, please create a reproducible example using a fork of [Cypress Test Tiny](https://github.com/cypress-io/cypress-test-tiny).  This gives us the best chance of seeing exactly what you are seeing and being able to investigate effectively.

**Comment by S-Lomax** [comment]
Hi @nagash77, thank you for your response. So far we've been unable to reproduce the issue using [https://github.com/cypress-io/cypress-test-tiny](url), even when we change the URL to our own app and use the same code. 

We've tried clearing the cypress cache and reinstalling cypress but still hit the same issue every time. Is there any other information we might be able to give you that would help?

**Comment by adrienharnay** [comment]
Not sure it is linked to a cache problem, as we are facing the same issue on our CI since upgrading from 12.14.0 to 12.17.3 and we don't have any cache on it. Haven't figured out a way to reproduce it yet either.

**Comment by bobilo-qa** [comment]
HI @nagash77 I have the same problem as mentioned above. We updated our cypress from 10 to the newest version and for some reason, this line just stopped working:   
```
uploadDocument() {
    cy.get(this.uploadDocumentsForm()).selectFile('cypress/fixtures/example.pdf', { force: true });
  }
```
I'm getting the same error: _must only be invoked from the spec file or support file_

**Comment by jbarthe** [comment]
Hi everyone, i'm still experiencing the same issue too. It was working on cypress 12.14.0 but since not.

### Cypress Version
12.17.3

### Test code to reproduce:

```
  it('should upload a document', () => {
    cy.intercept('POST', '/property_documents').as('uploadDocuments');

    cy.getBySel('dataroom-document-title')
      .first()
      .click();

    cy.getBySel('file-item').should('not.exist');

    cy.get('#file-uploader-1')
      .selectFile({
        contents: 'cypress/fixtures/rooms.json',
        fileName: 'file.png',
      }, { force: true });

    cy.wait('@uploadDocuments').then(() => {
      cy.getBySel('file-item').should('exist');
      cy.getBySel('dataroom-document-checked').should('exist');
      cy.getBySel('dataroom-document-unchecked').should('have.length', 21);
    });
  });

```

### Current behavior
![image](https://github.com/cypress-io/cypress/assets/9800317/f87d4479-a458-469c-b0f4-e15aff156aac)

**Comment by nagash77** [comment]
Hi @jbarthe at a glance that test *should* work.  Are you able to create a reproducible example using a fork of [Cypress Test Tiny](https://github.com/cypress-io/cypress-test-tiny).  This gives us the best chance of seeing exactly what you are seeing and being able to investigate effectively.

**Comment by BloodyRain2k** [comment]
I've started getting this issue a couple days ago. Completely randomly. Didn't update Cypress or anything.
Been working fine on this SPFX project using Cypress 12.17 for weeks.

And then I've started randomly getting these errors:
![Screenshot 2023-09-12 104116](https://github.com/cypress-io/cypress/assets/1014362/c85e332c-b3f9-482b-9078-62df6f092946)

The worst part being that they are literally random...
I can run the exact same test 20 times and 1-2 of those times it might throw it...

It would be really helpful if the error logger would include the invokation origin.
Because as far as I can see, it's always the same: the spec file where the code came from...

I have yet to even find out if that error above about the SP worker might be related.
Edit: ok, seems like that one's not related. Just had a run that was completely fine again, despite the Web Worker constantly throwing errors.

**Comment by jennifer-shehane** [comment]
@BloodyRain2k We'd really like to address this issue if it's happening unwarranted to users and randomly. 

Can anyone provide a reproducible example? Here are some tips for providing a [Short, Self Contained, Correct, Example](http://sscce.org/) and our own [Troubleshooting Cypress](https://on.cypress.io/troubleshooting) guide.

**Comment by S-Lomax** [comment]
Hi @jennifer-shehane we've been able to reproduce the issue using a fork of [](https://github.com/cypress-io/cypress/issues/url) but only using our application. We're currently trying to work on a solution where we can provide a successful reproduction to you without giving access to our application.

**Comment by newell-paul** [comment]
I have had this problem since attempting to update from 12.8.1 to 13.2.0. This tiny change fixed it in my particular case:

This line works in 12.8.1 and fails in 13.2.0 with the error: "must only be invoked from the spec file or support file."

`return cy.task('findDownloadMatching', pattern).then(cy.readFile)`

Changing the line to this works in 13.2.0:

`return cy.task('findDownloadMatching', pattern).then(filePath => {
      cy.readFile(filePath)
 })`

**Comment by KrzGruszcz** [comment]
@newell-paul I have the same problem: after upgrading Cypress to 13.3.1. getting this error: **must only be invoked from the spec file or support file.**

I was using cypress v11.2.0 and cy.readFile command like this:
```js
cy.wrap(<some_path>).then(cy.reafFile)
```

In v13.3.1 I have to change it to this:
```js
cy.wrap(<some_path>).then(path => cy.readFile(path))
```

**Comment by KrzysztofWisniewskiAcc** [comment]
Same problem on version 13.3.2 
More data:
https://github.com/cypress-io/cypress/issues/27845#issuecomment-1773932880

mb try to clean cache on your npm ?
I'm experiencing this on all environments, it's kind of strange that this is stated as 'occurs randomly '

**Comment by S-Lomax** [comment]
@jennifer-shehane, still working on getting a reproduction using [Cypress Test Tiny](https://github.com/cypress-io/cypress-test-tiny) but one thing we have noticed is that our site uses basic auth and when we bypass basic auth we no longer receive the error.

**Comment by S-Lomax** [comment]
Hi @jennifer-shehane / @nagash77 we've managed to reproduce using a fork from the cypress tiny test which can be accessed here:
[https://github.com/S-Lomax/cypress-test-tiny](url)

Please let me know if you need any further information.

**Comment by S-Lomax** [comment]
> @BloodyRain2k We'd really like to address this issue if it's happening unwarranted to users and randomly.
> 
> Can anyone provide a reproducible example? Here are some tips for providing a [Short, Self Contained, Correct, Example](http://sscce.org/) and our own [Troubleshooting Cypress](https://on.cypress.io/troubleshooting) guide.

@jennifer-shehane / @nagash77 bumping this thread again as we managed to reproduce using a fork from the cypress tiny test which can be accessed here:
[https://github.com/S-Lomax/cypress-test-tiny](url)

Looks like the cause of the issue in our case is when basic auth is on the site under test.

Please let me know if you need any further information.

**Comment by chrisbreiding** [comment]
@S-Lomax Good news! This issue has been fixed by https://github.com/cypress-io/cypress/pull/28428, which will be released in a new version of Cypress this week. Thank you for your patience!

**Comment by S-Lomax** [comment]
> @S-Lomax Good news! This issue has been fixed by #28428, which will be released in a new version of Cypress this week. Thank you for your patience!

Amazing, thanks for the update.
